Mod2/Lessons/Week1/VersionControlAndSoloGit.md

```diff
@@ -0,0 +1,113 @@
+
+Developers often have strict rules about when changes to a project can be made directly onto the `main` branch.  The `main` branch is typically reserved for production-ready code; `main` should always be ready to show to a client or stakeholder without fear of major bugs.  This means that new features are developed on tier own branches, and then **merged** into the main branch.
+
+When we create a new branch to work on a new feature, we call this **checking out a branch**.  We checkout a branch by creating a copy of `main` - that way our new feature is built with our current functionality as its base.
+
+Check out this visual representation of branching:
+![](/Mod2/Images/Week1/VCBranches.jpg)
+
+Here, we have a project that is created, then two branches are formed to work on different features (user and cart).  Once each feature is complete, it is merged into the `main` branch.  There may be many commits on each branch, and each of those commits will be represented in the commit history of the `main` branch when the feature branch is merged!  In this example, when the branches are merged, the commits will line up like this: 
+
+![](/Mod2/Images/Week1/VCBranchCommitLog.jpg)
+
+When we look at a commit log, we see the history of commits from newest to oldest - the most recent commit will be on top, and we can look backward by looking further down the list.
+
+> Individually, we are going to continue to practice working with git and branching by following along with the following 3 videos.  As before, pause the videos as necessary, and complete all steps demonstrated.  When finished, you will have created two branches, and merged those branches back to `main`.
+>1. [Creating Branches](https://www.loom.com/share/d675a00c7c634d35b977a7df6972b13c)
+>2. [Merging Branches](https://www.loom.com/share/d12d3d510da9402092fe793b4a398bd1)
+>3. [Merged Commit History](https://www.loom.com/share/e5baba208ec441a6a157a8b956dc30f9)
+
+After you have completed the individual work above:
+
+> In small groups: discuss the process that you just completed.  As a starter, answer the following questions:
+> * What is the purpose of branching?
+> * When we merge branches together, why might they not appear in chronological order in the commit history?
+> * What causes a merge conflict?
+
+** Instructor Note ** After letting groups discuss their challenges and questions, do a walkthrough of the process, answering questions as you go.
+
+### GitHub
+So far, we have created a project with a git repository that exists _only_ on our local machines.  This is fine for some personal projects, but most of the time, you are going to want to share your code.  When you share your code, you can get feedback and help a lot more easily.  Also, sharing your code allows for easier collaboration!
```

Comment:
This might be a good point to pose a question to students about why we would want to also have our projects on the cloud? And to draw parallels to google docs or google drive or dropbox (yay introducing google tools in mod 0).